### DISCUSSION TITLE
Personal bio site: 'firebase/auth' should be listed in the project's dependencies.

### DISCUSSION THREAD

**Discussion by Gabrielle-Tobermann** [nss-evening-cohort-14/evening-client-side#241] [Help Me!] [answered]
Gabby Tobermann 

Other tickets? No. 

Issue: I am trying to setup auth for the portfolio project. I am using the class React template. When trying to import firebase, I get the following error:
 'firebase/auth' should be listed in the project's dependencies. Run 'npm i -S firebase/auth' to add it  import/no-extraneous-dependencies

I checked the package.json file and firebase is listed as a dependency. I tried to run ```npm i -S firebase/auth``` but it didn't work (since it's already listed). I tried stopping and restarting my server and VS code but the error is still there. 

What I googled: 
https://firebase.google.com/docs/functions/handle-dependencies

Link to code: 
https://github.com/Gabrielle-Tobermann/react-personal-bio-site/blob/setup/src/App/index.js
https://github.com/Gabrielle-Tobermann/react-personal-bio-site/blob/setup/package.json

**Comment by drteresavasquez** [comment]
Did you add `import 'firebase/auth'` to the top of the file that is using the auth method?

**Reply by drteresavasquez** [reply]
Never mind. I see your code and you did 🤦🏽‍♀️. Try deleting your node modules and run non install again. Also make sure your rules are correct. 

And move line 10 to your index.js making sure to import Firebase.

**Reply by Gabrielle-Tobermann** [reply]
So I exported my json data and re-did my firebase project completely, with no luck. I checked the rules and I don't see anything wrong with them. Then I restarted a new repo with the react template, hoping that I had just missed a step in the setup but the error is still there. I moved line 10 to the other index.js. I ran npm install multiple times, with no success. I checked the .env file and the keys are correct. I closed and restarted VS code as well. I tried running the command listed in the error: ```npm i -S firebase/app``` but it didn't help. I also tried running ```npm install --save firebase``` which I found in the firebase docs but that didn't help either. I will continue to investigate but I will probably need to ask for help with this tomorrow during class.

**Comment by drteresavasquez** [comment]
Use this in App/index.js

import firebase from 'firebase/app';
import 'firebase/auth';

firebase.auth().onAuthStateChanged((authed) => {...}

Use this in src/index.js

```javascript 
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import firebase from 'firebase/app';
import firebaseConfig from './helpers/apiKeys';
import App from './App';
import reportWebVitals from './reportWebVitals';

import 'bootstrap/dist/css/bootstrap.min.css';
import './styles/index.scss';

firebase.initializeApp(firebaseConfig);

ReactDOM.render(
  <React.Fragment>
    <Router>
      <App />
    </Router>
  </React.Fragment>,
  document.getElementById('root')
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();

```

If none of those work then you may not have set up authentication properly in Firebase. We can take a look for sure.

**Reply by Gabrielle-Tobermann** [reply]
I made the changes from above but the error is still there.

**Reply by drteresavasquez** [reply]
Yeah. I think there were some updates to the API. We discussed in group how to fix, so all should be good now! Bravo!

**Comment by Gabrielle-Tobermann** [comment, marked as answer]
Solution: Firebase may have updated some stuff, which is why the imports were not working. Instead of ```import firebase from 'firebase/app';``` it should be ```import firebase from 'firebase';``` and delete the firebase/auth import.

**Reply by jimguistical** [reply]
Thank you for this!
